x-pack/plugins/lens/public/indexpattern_datasource/datapanel.tsx

```diff
@@ -488,23 +561,84 @@ export const InnerIndexPatternDataPanel = function InnerIndexPatternDataPanel({
         responsive={false}
       >
         <EuiFlexItem grow={false}>
-          <div className="lnsInnerIndexPatternDataPanel__header">
-            <ChangeIndexPattern
-              data-test-subj="indexPattern-switcher"
-              trigger={{
-                label: currentIndexPattern.title,
-                title: currentIndexPattern.title,
-                'data-test-subj': 'indexPattern-switch-link',
-                fontWeight: 'bold',
-              }}
-              indexPatternId={currentIndexPatternId}
-              indexPatternRefs={indexPatternRefs}
-              onChangeIndexPattern={(newId: string) => {
-                onChangeIndexPattern(newId);
-                clearLocalState();
-              }}
-            />
-          </div>
+          <EuiFlexGroup
+            gutterSize="s"
+            alignItems="center"
+            className="lnsInnerIndexPatternDataPanel__header"
+          >
+            <EuiFlexItem grow={true} className="lnsInnerIndexPatternDataPanel__switcher">
+              <ChangeIndexPattern
+                data-test-subj="indexPattern-switcher"
+                trigger={{
+                  label: currentIndexPattern.title,
+                  title: currentIndexPattern.title,
+                  'data-test-subj': 'indexPattern-switch-link',
+                  fontWeight: 'bold',
+                }}
+                indexPatternId={currentIndexPatternId}
+                indexPatternRefs={indexPatternRefs}
+                onChangeIndexPattern={(newId: string) => {
+                  onChangeIndexPattern(newId);
+                  clearLocalState();
+                }}
+              />
+            </EuiFlexItem>
+            {addField && (
+              <EuiFlexItem grow={false}>
+                <EuiPopover
+                  panelPaddingSize="s"
+                  isOpen={popoverOpen}
+                  closePopover={() => {
+                    setPopoverOpen(false);
+                  }}
+                  data-test-subj="lnsIndexPatternActions-popover"
+                  button={
+                    <EuiButtonIcon
+                      color="text"
```

Comment:
Add an `aria-label` to icon buttons to make the accessible to screen readers

```suggestion
                      aria-label='Index pattern settings' // TODO i18n 
                      color="text"
```